From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Thu, 1 Dec 2022 16:43:50 +0100
Subject: [PATCH] Do not log run as root warning

Ported from Winds-Studio/Leaf ver/26.2 (0035-Do-not-log-run-as-root-warning.patch), GPL-3.0-only.
Original author: Martijn Muijsers.
Canvas adaptation: gale.log.warning.root system property mapped to logs.disableRootWarning (default true).

diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index d185521ed69d3908196c731e66e5fde27edb9785..f83ed15b8559be420b4d7795b8c8e5333fbc37af 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -211,7 +211,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         }
 
         // Paper start - detect running as root
-        if (io.papermc.paper.util.ServerEnvironment.userIsRootOrAdmin()) {
+        if (!io.canvasmc.canvas.GlobalConfiguration.getInstance().logs.disableRootWarning && io.papermc.paper.util.ServerEnvironment.userIsRootOrAdmin()) { // Canvas - KeYi - do not log run as root warning
             LOGGER.warn("****************************");
             LOGGER.warn("YOU ARE RUNNING THIS SERVER AS AN ADMINISTRATIVE OR ROOT USER. THIS IS NOT ADVISED.");
             LOGGER.warn("YOU ARE OPENING YOURSELF UP TO POTENTIAL RISKS WHEN DOING THIS.");
